julia/src/api_types.jl

```diff
@@ -33,22 +33,35 @@ function bust_union(x::TaggedUnion{U}) where U
 end
 bust_union(x) = x  # not a union, leave it alone
 
+"Forces reinterpretting `raw` as a `T`. If sizeof(raw) <: size(T) high bits will be filled with junk."
 function force_reinterpret(::Type{T}, raw) where T
     isbits(raw) || throw(ArgumentError("Can only reinterpret from a isbits type"))
     isbitstype(T) || throw(ArgumentError("Can only reinterpret into a isbits type"))
     return unsafe_load(Ptr{T}(Base.pointer_from_objref(Ref(raw))))
 end
 
 
-struct CLit <: TaggedUnion{Tuple{Int64, Int32, Int8, Float64, Float32, UInt64, UInt32}}
+struct CLit <: TaggedUnion{Tuple{Int64, Int32, Int8, Float64, Float32, UInt32, UInt64}}
     tag::UInt64
     payload::NTuple{8, UInt8}  # actually the Union, needs to be big enough to hold largest which is an Int64
 end
+CLit(x) = CLit(CLit_tag(x), force_reinterpret(NTuple{8, UInt8}, x))
+
+CLit_tag(::Int64) = 0x0000_0000_0000_0000
+CLit_tag(::Int32) = 0x0000_0000_0000_0001
+CLit_tag(::Int8) = 0x0000_0000_0000_0002
```

Comment:
This is UInt8 by the way. I think I forgot that there was this one small bugfix.

Reply:
This should also be updated in the `CLit` struct definition above.